Tools/ros2/ardupilot_sitl/launch/sitl.launch.py

```diff
@@ -0,0 +1,184 @@
+from launch import LaunchDescription
+from launch.actions import DeclareLaunchArgument
+from launch.actions import ExecuteProcess
+from launch.actions import OpaqueFunction
+from launch.substitutions import LaunchConfiguration
+from launch.substitutions import PathJoinSubstitution
+
+from launch_ros.substitutions import FindPackageShare
+
+# labels for the optional uart launch arguments
+uart_labels = ["A", "B", "C", "D", "E", "F", "H", "I", "J"]
+
+
+def launch_sitl(context, *args, **kwargs):
+    """
+    Launch SITL
+    """
+    command = LaunchConfiguration("command").perform(context)
+    model = LaunchConfiguration("model").perform(context)
+    speedup = LaunchConfiguration("speedup").perform(context)
+    slave = LaunchConfiguration("slave").perform(context)
+    sim_address = LaunchConfiguration("sim_address").perform(context)
+    instance = LaunchConfiguration("instance").perform(context)
+    defaults = LaunchConfiguration("defaults").perform(context)
+
+    # debug
+    print(f"command:          {command}")
+    print(f"model:            {model}")
+    print(f"speedup:          {speedup}")
+    print(f"slave:            {slave}")
+    print(f"sim_address:      {sim_address}")
+    print(f"instance:         {instance}")
+    print(f"defaults:         {defaults}")
+
+    # required arguments
+    cmd_args = [
+        f"{command} ",
+        f"--model {model} ",
+        f"--speedup {speedup} ",
+        f"--slave {slave} ",
+        f"--sim-address={sim_address} ",
+        f"--instance {instance} ",
+        f"--defaults {defaults} ",
+    ]
+
+    # optional arguments
+    wipe = LaunchConfiguration("wipe").perform(context)
+    if wipe == "True":
+        cmd_args.append("--wipe ")
+        print(f"wipe:             {wipe}")
+
+    synthetic_clock = LaunchConfiguration("synthetic_clock").perform(context)
+    if synthetic_clock == "True":
+        cmd_args.append("--synthetic-clock ")
+        print(f"synthetic_clock:  {synthetic_clock}")
+
+    home = LaunchConfiguration("home").perform(context)
+    if home:
+        cmd_args.append("--home {home} ")
+        print(f"home:             {home}")
+
+    # optional uart arguments
+    for label in uart_labels:
+        arg = LaunchConfiguration(f"uart{label}").perform(context)
+        if arg:
+            cmd_args.append(f"--uart{label} {arg} ")
+            print(f"uart{label}:            {arg}")
+
+    # optional serial arguments
+    for label in range(10):
```

Comment:
```suggestion
    for label in range(len(uart_labels)+1):
```

Avoid magic numbers please

Reply:
Introduced variable `max_serial_ports`.

Changed in f60fd749f7